### PR TITLE
Add ARA records ansible

### PR DESCRIPTION
**Pull request by OgarOgarovic**
Adds the option to deploy [ARA records ansible](https://ara.recordsansible.org/) recording all ansible output on the host and also inside VMs, providing a HTTP UI on the host.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,6 +57,19 @@ pip install lxml
 
 ### Building Environment
 The `builder` script is used to manage nodes with `ansible-playbook` .
+#### Deploying ARA records ansible server
+
+To deploy ARA server and start recording host ansible plays with ensure flag
+`ara_enable` in `globals.yml` is set to true and run:
```
Comment (MatusJenca2) [inline README.md]: I don't think `globals.yml` is where ARA should be configured.
`globals.yml` is for kolla-ansible configuration.

File: README.md [hunk 2]
```diff
@@ -56,6 +57,19 @@ pip install lxml
...
+
+```bash
+./builder ara user/local-aio.yml
+```
```
Comment (MatusJenca2) [inline README.md:67]:
```
    The conditional check 'ara_enable | bool' failed. The error was: error while evaluating conditional (ara_enable | bool): 'ara_enable' is undefined
```
Please add default value to `group_vars: all`

File: spawn.yml [hunk 3]
```diff
@@ -9,3 +9,45 @@
 
     - name: create_multinode
       when: not is_aio
+
+  tasks:
```
Comment (MatusJenca2) [inline spawn.yml]: I dont understand why you put this here and not prepare playbook.

File: spawn.yml [hunk 4]
```diff
@@ -9,3 +9,45 @@
...
+        hostname: "{{ item }}"
+        ansible_private_key_file: "{{ kolla_ssh_key }}"
+        group: ara_deployments
+      loop: "{{ deployment_hosts }}"
+
+- name: Configure ARA recording on deployment nodes
+  hosts: ara_deployments
+  user: kolla
+  gather_facts: false
+
```
Comment (MatusJenca2) [inline spawn.yml]:
```suggestion
    vars:
      ansible_ssh_common_args: '-o IdentitiesOnly=yes'
      
```
Or just move all this to `prepare` playbook.

File: roles/config_ara_recording/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  ansible.builtin.lineinfile:
+    path: "{{ config_to_edit }}"
+    regexp: '^api_server'
+    insertafter: '^\[ara\]'
+    line: "api_server = http://{{ ara_server_ip }}:{{ ara_server_port }}"
+    state: present
+
+- name: Get ARA callback plugins definition
+  ansible.builtin.shell: |
+    . {{ (python_bin_path, 'activate') | path_join }}
```
Comment (MatusJenca2) [inline roles/config_ara_recording/tasks/main.yml]: This works only in a virtual environment, not with natively installed python i.e. `/usr/bin/python`
```
TASK [config_ara_recording : Get ARA callback plugins definition] 
fatal: [localhost]: FAILED! => changed=true 
  cmd: |-
    . /usr/bin/activate
    python3 -m ara.setup.callback_plugins
  delta: '0:00:00.002860'
  end: '2024-05-29 13:22:11.322442'
  msg: non-zero return code
  rc: 127
  start: '2024-05-29 13:22:11.319582'
  stderr: '/bin/sh: 1: .: Can''t open /usr/bin/activate'
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>

```

File: ara.yml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+---
```
Comment (MatusJenca2) [inline ara.yml:1]: How does this work on a remote libvirt host?

**Review by MatusJenca2 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (OgarOgarovic) [reply README.md]: Wrong info in README, should be user_config, I'll fix it.

File: README.md (hunk 2 above)
Comment (OgarOgarovic) [reply README.md:67]: Added...

File: spawn.yml (hunk 3 above)
Comment (OgarOgarovic) [reply spawn.yml]: Valid point, moved it to prepare playbook.

File: spawn.yml (hunk 4 above)
Comment (OgarOgarovic) [reply spawn.yml]: Moved it to prepare playbook.

File: roles/config_ara_recording/tasks/main.yml (hunk 5 above)
Comment (OgarOgarovic) [reply roles/config_ara_recording/tasks/main.yml]: Removed the dependency on virtual environment activation.

File: ara.yml (hunk 6 above)
Comment (OgarOgarovic) [reply ara.yml:1]: Updated to properly install on local or remote host.

File: spawn.yml [hunk 7]
```diff
@@ -9,3 +9,45 @@
...
+#       ansible.builtin.setup:
+#
+#   roles:
+#     - role: config_ara_recording
+#       vars:
+#         # TODO: Better solution to get the controllers IP
+#         ara_server_ip: "{{ ansible_facts['env']['SSH_CLIENT'].split() | first }}"
+#         config_to_edit: "~/ansible.cfg"
+#         python_bin_path: "~/kolla/bin/"
+#       when: ara_enable | bool
```
Comment (MatusJenca2) [inline spawn.yml]: Fine, just remove the commented code and we can merge.

**Review by MatusJenca2 — CHANGES_REQUESTED** [review]

**Pull request merged**